libs/framework/src/service_registry.c

```diff
@@ -1124,12 +1044,7 @@ celix_status_t celix_serviceRegistry_addServiceListener(celix_service_registry_t
         event.reference = ref;
         event.type = OSGI_FRAMEWORK_SERVICE_EVENT_REGISTERED;
         listener->serviceChanged(listener->handle, &event);
```

Comment:
Consider when serviceTracker_invokeAddingService returns error, the newly created reference might need to be removed from `registry->serviceReferences`.